src/main/java/hudson/plugins/gradle/injection/InjectionConfig.java

```diff
@@ -494,11 +588,38 @@ private static FormValidation checkVersion(String value, boolean required) {
      * Invoked by XStream when this object is read into memory.
      */
     @SuppressWarnings("unused")
-    protected Object readResolve() {
+    protected Object readResolve() throws IOException {
         if (injectionVcsRepositoryPatterns != null) {
             String filters = migrateLegacyRepositoryFilters(injectionVcsRepositoryPatterns);
             parsedVcsRepositoryFilter = VcsRepositoryFilter.of(filters);
         }
+        if (accessKey != null && accessKeyCredentialId == null) {
```

Comment:
This function will always be called, yes. I think `accessKey` will be saved as null afterwards, as there are not getters and setters for it anymore, so serialization would probably skip it. But I can add the `acceessKey=null` just in case.